From de2f8e52b2dbeed96a9d003a18051fcccb4624ab Mon Sep 17 00:00:00 2001
From: Maksym Komar <komar@evologics.de>
Date: Tue, 16 Jan 2018 15:36:01 +0100
Subject: [PATCH 1/2] fix tupo with lost struct spi_flash

---
 driver/spi_flash/spi_flash.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/driver/spi_flash/spi_flash.c b/driver/spi_flash/spi_flash.c
index 5cbb9bd..13cc58a 100644
--- a/driver/spi_flash/spi_flash.c
+++ b/driver/spi_flash/spi_flash.c
@@ -209,7 +209,7 @@ int spi_flash_setup(struct spi_flash *flash,
 }
 
 #if defined(CONFIG_LOAD_LINUX) || defined(CONFIG_LOAD_ANDROID)
-static int update_image_length(spi_flash_t *flash,
+static int update_image_length(struct spi_flash *flash,
 			       unsigned int offset,
 			       unsigned char *dest,
 			       unsigned char flag)
-- 
2.15.1

