diff --git a/arch/arm/cpu/armv7/mx6/clock.c b/arch/arm/cpu/armv7/mx6/clock.c
index c3c9c5e..605aae2 100644
--- a/arch/arm/cpu/armv7/mx6/clock.c
+++ b/arch/arm/cpu/armv7/mx6/clock.c
@@ -350,7 +350,7 @@ static u32 decode_pll(enum pll_clocks pll, u32 infreq)
 		post_div = 1 << (2 - post_div);
 		div &= BM_ANADIG_PLL_AUDIO_DIV_SELECT;
 
-		freq = (u64)infreq * pll_num / pll_denom;
+		freq = lldiv((u64)infreq * pll_num, pll_denom);
 		freq += infreq * div;
 		return lldiv(freq, post_div);
 	case PLL_VIDEO:
@@ -371,7 +371,7 @@ static u32 decode_pll(enum pll_clocks pll, u32 infreq)
 		post_div = 1 << (2 - post_div);
 		div &= BM_ANADIG_PLL_VIDEO_DIV_SELECT;
 
-		freq = (u64)infreq * pll_num / pll_denom;
+		freq = lldiv((u64)infreq * pll_num, pll_denom);
 		freq += infreq * div;
 		return lldiv(freq, post_div);
 	case PLL_ENET:
